app/scripts/lib/setupAccountLabelsPetnamesBridge.ts

```diff
@@ -0,0 +1,46 @@
+import { NameController, NameType } from '@metamask/name-controller';
+import {
+  PreferencesController,
+  AccountIdentityEntry,
+  PreferencesControllerState,
+} from '../controllers/preferences';
+import { NameEntry, OneWayNameBridge } from './NameBridge';
+
+export const ACCOUNT_LABEL_NAME_TYPE = NameType.ETHEREUM_ADDRESS;
+export const ACCOUNT_LABEL_VARIATION = '*';
+
+/**
+ * Sets up a bridge between the account labels in the preferences controller and
+ * the petnames in the name controller.
+ *
+ * @param preferencesController - The preferences controller to listen to.
+ * @param nameController - The name controller to update.
+ */
+export default function setupAccountLabelsPetnamesBridge(
```

Comment:
If we use an abstract class, then this entire file would become a child class such as `AccountIdentitiesPetnamesBrdige` and we'd register the listeners using the `init` method provided by `AbstractPetnamesBridge`.